Cli updates

logan-anderson opened the pull request:
# Separates the init into a backend init and normal init.


## Changes

## `tinacms init`
- sets all all files required to use Tina Locally
- does not ask for clientID or token but instead will add `process.env.**`

## `tinacms init backend`
- If the user has not setup tina yet it will setup tina
- Asks the user if they want to use Tina Cloud or self host
-  if they want to use Tina Cloud it will ask for the clientID and token and populate the .env file
- If they use Self hosted tina (and use Next.js) it will add all files and env vars required to setup self hosting

## Can test by
`npx @tinacms/cli@cli-updates init backend --tinaVersion cli-updates`

jamespohalloran commented:
I tested `npx @tinacms/cli@cli-updates init backend --tinaVersion cli-updates` and chose "Tina Cloud". It doesn't look like it did anything (no changes to the .env)

jamespohalloran commented:
@logan-anderson At the end of tina init backend, I had to scroll up to see the instructions about inserting the code snippet to the tina config:

![Screenshot 2023-08-16 at 9 35 42 AM](https://github.com/tinacms/tinacms/assets/3323181/72bd604e-edac-4a35-9685-0c2d874b740e)



![Screenshot 2023-08-16 at 9 35 29 AM](https://github.com/tinacms/tinacms/assets/3323181/54466e55-ff8b-4838-9d0b-9b6b74595e04)

I think some users will miss it. 
We also say: 
> You can now run your build command and deploy your site.

Maybe we should log the code snippet after all the `Adding file at tina/database.js...` lines? Also, is there a CLI style we can add to it so make the code-snippet stand out more?

logan-anderson commented:
@jamespohalloran I have fixed all of the issues you have found. I also updated the styling but not sure if I am totally happy with it.

> @logan-anderson I tested `npx @tinacms/cli@cli-updates init backend --tinaVersion cli-updates` and chose "Tina Cloud". It doesn't look like it did anything (no changes to the .env)

This should add the values to the .env.tina file. Maybe we should add a message telling the user they can copy those values into there .env file?

jamespohalloran commented on packages/@tinacms/cli/src/cmds/init/configure.ts:
```diff
@@ -79,19 +79,236 @@ async function configure(
+      message: `What is your GitHub Personal Access Token? (Hit enter to skip and set up later)\n${logText(
+        'Learn more here: '
+      )}${linkText(
+        'https://tina.io/docs/self-hosted/starters/nextjs-vercel/'
```
@logan-anderson This should probably link to the /docs/self-hosted/existing-site/ link instead of the starter doc. (It looks like that doc may need to be updated to have some PAT context though)

logan-anderson replied:
Updated the link.

jamespohalloran commented on packages/@tinacms/cli/src/cmds/init/apply.ts:
```diff
@@ -178,27 +161,90 @@ async function apply({
+  ) {
+    // if we are doing a backend init we should print out what they need to add to the config
+    logger.info(
+      'Please add the following to your tina config:\n' +
```
```suggestion
      '\nPlease add the following to your tina config:\n' +
```

jamespohalloran commented on packages/@tinacms/cli/src/cmds/init/configure.ts:
```diff
@@ -79,19 +86,234 @@ async function configure(
+    },
+    {
+      name: 'dataLayerAdapter',
+      message: 'Select a self-hosted data layer adapter',
```
We kind of use "Database Adapter" and "Data layer adapter" in the docs. Is one correct?

logan-anderson replied:
I think we should stick with one. Lets go with Database adapter for now and we can update it if we need.

jamespohalloran commented on packages/@tinacms/cli/src/cmds/init/apply.ts:
```diff
@@ -178,27 +161,90 @@ async function apply({
+  ) {
+    // if we are doing a backend init we should print out what they need to add to the config
+    logger.info(
+      '\nPlease add the following to your tina config:\n' +
```
Might want to specify the file that this needs to go in

jamespohalloran commented on packages/@tinacms/cli/src/cmds/init/configure.ts:
```diff
@@ -79,19 +86,234 @@ async function configure(
+      },
+      message: `What is your GitHub Personal Access Token? (Hit enter to skip and set up later)\n${logText(
+        'Learn more here: '
+      )}${linkText('https://tina.io/docs/self-hosted/existing-site/')}`,
```
```suggestion
      )}${linkText('https://tina.io/docs/self-hosted/existing-site/#github-personal-access-token')}`,
```

logan-anderson commented:
@jamespohalloran updated the text based on your feedback
